app/routes/goal_route.py

```diff
@@ -0,0 +1,101 @@
+from flask import Blueprint, jsonify, request
+from app.models.goal import Goal
+from app.models.task import Task
+from app import db
+from app.routes.task_route import validate_model
+
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+# route to get all goals
+
+@goals_bp.route("", methods=["GET"])
+def read_all_goals():
+
+    goals = Goal.query.all()
+
+    goal_response = []
+    for goal in goals:
+        goal_response.append(goal.to_dict())
+    return jsonify(goal_response), 200
+
+# route to create a new goal
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+
+    if request_body.get("title"):
+        new_goal = Goal(title=request_body["title"])
+
+        db.session.add(new_goal)
```

Comment:
Great job using `db.session.<method-name>`. Session gives us access to the follow:

- `db` is our app’s instance of SQLAlchemy.
- `session` represents our active database connection.
- By referencing `db.session` we can use SQLAlchemy’s methods to perform tasks like:
    - committing a change to a model
    - storing a new record of a model
    - deleting a record.
- By referencing `db.session.add()` you are able to use the SQLAlchemy method to store a new record of the task model